sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2FunctionSuite.scala

```diff
@@ -466,6 +526,60 @@ class DataSourceV2FunctionSuite extends DatasourceV2SQLBase {
     override def produceResult(state: (Long, Long)): Long = state._1 / state._2
   }
 
+  object UnboundDecimalAverage extends UnboundFunction {
+    override def name(): String = "favg"
+
+    override def bind(inputType: StructType): BoundFunction = {
+      if (inputType.fields.length > 1) {
+        throw new UnsupportedOperationException("Too many arguments")
+      }
+
+      // put interval type here for testing purpose
+      inputType.fields(0).dataType match {
+        case _: NumericType | _: DayTimeIntervalType => DecimalAverage
+        case dataType =>
+          throw new UnsupportedOperationException(s"Unsupported input type: $dataType")
+      }
+    }
+
+    override def description(): String =
+      """iavg: produces an average using decimal division, ignoring nulls
```

Comment:
Just a question. What is the meaning of `i` here? Previously, I thought it stands for `Integer`.
BTW, `davg` may look a little confusing because it can be read as DecimalAverage or DoubleAverage.

Reply:
my bad - let me fix the name to be something more meaningful, like `decimal_avg`.